### PR TITLE
chore(publish): fix dev dep install on docgen when publishing

### PR DESCRIPTION
**Pull request by bobylito**
Also adds `yarn install` before building instantsearch.js for the website

### CODE REVIEW

**Comment by bobylito** [comment]
NB: it doesn't work with yarn 1.1.0. But it works with 1.2.0

**Comment by vvo** [comment]
https://github.com/algolia/instantsearch.js/blob/2208f7d93c5c1fa906b3782764fd9ed5de006f09/scripts/docs/publish.sh#L29

Since the publish scripts sets NODE_ENV to production this will not install devDependencies no?

We could force NODE_ENV to development if that's the case in build.sh

**Comment by bobylito** [comment]
It's updated @vvo

File: scripts/docs/build-dev.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #! /usr/bin/env bash
 
-set -ev # exit when error
+set -e # exit when error
```
Comment (iam4x) [inline scripts/docs/build-dev.sh]: Just for information, what is `v` stand for?

Comment (vvo) [reply scripts/docs/build-dev.sh]: verbose, every line of the script will be also written on the output (not just the effect, but also the line itself).

doc: http://tldp.org/LDP/abs/html/options.html (not easy to find!)

**Pull request merged**